downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -89,41 +90,55 @@ open class GitWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(work
     }
 
     override fun getNested(): Map<String, VcsInfo> =
-        listSubmodulePaths(repo).associateWith { GitWorkingTree(repo.workTree.resolve(it), vcsType).getInfo() }
+        repo.use {
```

Comment:
Have you considered exposing the same mechanism as the Repository class, e.g. `open` and `close()` functions instead? I believe with such approach it would be easier to move towards thread safety (hm, may we need that in the future? if not irrelevant). Note that not all subclasses of `Repository` are thread safe but the `FileRepository` is.

Reply:
> Have you considered exposing the same mechanism as the Repository class, e.g. incrementOpen and close() functions instead?

No, I'd prefer to not reimplement that logic here.

> Note that not all subclasses of `Repository` are thread safe but the `FileRepository` is.

Aren't we good then already in terms of thread safety? Anyway, the goal of this PR is not thread-safety, but to avoid a resource leak. Let's do one topic at a time.

Reply:
> Anyway, the goal of this PR is not thread-safety, but to avoid a resource leak. Let's do one topic at a time.

I was a bit unclear. My goal was not to ask you to put thread safety in scope of the PR. My intention was rather to bring it to attention and to clarify if we think it will be needed in future. Anyhow, I'm convinced now that it won't make moving towards thread safety (much) more difficult...